consensus/src/dag/reconstruction/parents.rs

```diff
@@ -178,6 +180,7 @@ impl<U: Unit> Reconstruction<U> {
                             unit_hash,
                             parent_coord.creator(),
                             *parent_hash,
+                            parent_coord.round(),
```

Comment:
While writing some unit test to cover scenario in which a unit pretend to have different coord - there are two distinc sources of information passed now down to `with_parents:
1. `parent_hash` from the line above comes from `units_by_coord` map, which is filled what a unit claims to have its coord
2. `parent_coord.round` is really taken from its control hash
and since we don't have any validation between unit coord and its control hash, at least in the tests I can pass here mismatched data, ie see commit https://github.com/Cardinal-Cryptography/AlephBFT/pull/514/commits/aef31c8892f6aa4345c75729986ca367ce91afc6 . Now my question is, is this line okay, or should I change `units_by_coord` to keep both hash and round, so sources of both value is the same?

Reply:
This comment is not valid anymore, since I did not take into account validation, which filters this scenario